File: Leetcode/941/941.cpp
class Solution {
public:
    bool strictlyIncrease(const vector<int> &arr, int l, int r) {
        if (r - l == 0) return false;
        if (r - l == 1) return true;
        for (int i = l; i < r; ++i) {
            if (arr.at(i) >= arr.at(i + 1)) {
                return false;
            }
        }
        return true;
    }
    bool strictlyDecrease(const vector<int> &arr, int l, int r) {
        if (r - l == 0) return false;
        if (r - l == 1) return true;
        for (int i = l; i < r; ++i) {
            if (arr.at(i) <= arr.at(i + 1)) {
                return false;
            }
        }
        return true;
    }
    bool validMountainArray(vector<int>& arr) {
        if (arr.size() < 3) return false;
        auto iter = max_element(arr.begin(), arr.end());
        if (count(arr.begin(), arr.end(), *iter) != 1) return false;
        int indexOfMax = iter - arr.begin();
        return strictlyIncrease(arr, 0, indexOfMax) && strictlyDecrease(arr, indexOfMax, (int)arr.size()-1);
    }
};

File: Leetcode/941/941.go
func validMountainArray(arr []int) bool {
	if len(arr) < 3 {
		return false
	}
	maxValue := -1
	index := 0
	for i := 0; i < len(arr); i++ {
		if maxValue < arr[i] {
			maxValue = arr[i]
			index = i
		}
	}
	if index == 0 || index == len(arr) - 1 {
		return false
	}
	for i := index; i >= 1; i-- {
		if arr[i] <= arr[i-1] {
			return false
		}
	}
	for i := index; i < len(arr)-1; i++ {
		if arr[i] <= arr[i+1] {
			return false
		}
	}
	return true
}